osu.Game/Screens/Select/BeatmapCarousel.cs

```diff
@@ -618,27 +617,32 @@ protected override void Update()
                     // Add those items within the previously found index range that should be displayed.
                     foreach (var item in toDisplay)
                     {
-                        var panel = setPool.Get(p => p.Item = item);
+                        var panel = setPool.Get();
 
+                        panel.Item = item;
                         panel.Depth = item.CarouselYPosition;
-                        panel.Y = item.CarouselYPosition;
+                        panel.UpdateY();
 
                         Scroll.Add(panel);
                     }
                 }
             }
 
+            // Update Y positions for the non-drawable models
+            if (Clock != null)
```

Comment:
How can clock ever be null? (hint: it can't)

Reply:
It can in testing. The testing only function somehow calls update before the clock is initialized. Could try to remove that function instead.

https://github.com/ppy/osu/blob/941ccb6939bea605132e9818b532212fb86ce6e6/osu.Game/Screens/Select/BeatmapCarousel.cs#L98-L103

Reply:
That just exposes the root of the problem which is coupling of models to their drawable representation, which [I warned about in discord yesterday already](https://discord.com/channels/188630481301012481/188630652340404224/787445202310529055).